pycbc/conversions.py

```diff
@@ -395,6 +425,107 @@ def secondary_spin(mass1, mass2, spin1, spin2):
     ss[mask] = spin1[mask]
     return _formatreturn(ss)
 
+def primary_xi(spinx, spiny):
+    """Returns the effective precession spin argument for the larger mass.
+    """
+    return chi_perp_from_spin1x_spin1y(spinx, spiny)
+
+def secondary_xi(mass1, mass2, spinx, spiny):
+    """Returns the effective precession spin argument for the smaller mass.
+    """
+    q = q_from_mass1_mass2(mass1, mass2)
+    a1 = 2 + 3 * q / 2
+    a2 = 2 + 3 / (2 * q)
+    return a1 / (q**2 * a2) * chi_perp_from_spin1x_spin1y(spinx, spiny)
+
+def chi_perp_from_spin1x_spin1y(spin1x, spin1y):
+    """Returns the in-plane spin from spin1x and spin1y.
+    """
+    return numpy.sqrt(spin1x**2 + spin1y**2)
+
+def chi_perp_from_spin2x_spin2y(spin2x, spin2y):
+    """Returns the in-plane spin from spin2x and spin2y.
+    """
+    return chi_perp_from_spin1x_spin1y(spin2x, spin2y)
+
+def chi_perp_from_mass1_mass2_xi2(mass1, mass2, xi2):
+    """Returns the in-plane spin from mass1, mass2, and xi2 for the
+    secondary mass.
+    """
+    q = q_from_mass1_mass2(mass1, mass2)
+    a1 = 2 + 3 * q / 2
+    a2 = 2 + 3 / (2 * q)
+    return q**2 * a2 / a1 * xi2
+
+def chi_p_from_xi1_xi2(xi1, xi2):
+    """Returns effective precession spin from xi1 and xi2.
+    """
+    xi1 = _ensurearray(xi1)
+    xi2 = _ensurearray(xi2)
+    if xi1.shape != xi2.shape:
+        raise ValueError("xi1, xi2 must have same shape")
+    chi_p = copy.copy(xi1)
+    mask = xi1 < xi2
+    chi_p[mask] = xi2[mask]
+    return _formatreturn(chi_p)
+
+def phi1_from_phi_a_phi_s(phi_a, phi_s):
+    """Returns the angle between the x-component axis and the in-plane
+    spin for the primary mass from phi_s and phi_a.
+    """
+    return (phi_s + phi_a) / 2.0
+
+def phi2_from_phi_a_phi_s(phi_a, phi_s):
+    """Returns the angle between the x-component axis and the in-plane
+    spin for the secondary mass from phi_s and phi_a.
+    """
+    return (phi_s - phi_a) / 2.0
+
+def phi1_from_spin1x_spin1y(spin1x, spin1y):
+    """Returns phi1 for primary mass.
+    """
+    return numpy.arctan(spin1y / spin1x)
+
+def phi2_from_spin2x_spin2y(spin2x, spin2y):
+    """Returns phi2 for secondary mass.
+    """
+    return phi1_from_spin1x_spin1y(spin2x, spin2y)
```

Comment:
Same comment as above... just name this `phi_from_spinx_spiny` and get rid of the second function.